website/frontend/reduxStore/Press/PressSlice.js

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { getAllPressApi } from '../../apis';
+
+const initialState = {
+  loading: false,
+  press: [],
+  error: null
+};
+
+const pressSlice = createSlice({
+  name: 'press',
+  initialState,
+  reducers: {
+    loadPressSuccess(state, action) {
+      state.press = action.payload;
+      state.loading = false;
+      state.error = null;
+    },
+    loadPressFailure(state, action) {
+      state.loading = false;
+      state.error = action.payload;
+    },
+    startLoading(state) {
+      state.loading = true;
+    }
+  }
+});
+
+export const { loadPressSuccess, loadPressFailure, startLoading } = pressSlice.actions;
+
+export default pressSlice.reducer;
+
+export const loadPressData = () => async (dispatch) => {
+  try {
+    dispatch(startLoading());
+    const resData = await getAllPressApi();
+    dispatch(loadPressSuccess(resData));
+  } catch (err) {
+    dispatch(loadPressFailure(err.message));
+  }
+};
```

Comment:
Great work with refactoring @OchiengPaul442 

Given that Javascript is an interpreted language, it's always good practice to have logic functions at the top of the page such that nothing breaks later on. I'll have this refactored then clear the PR for merge

Reply:
Okay, @FaithDaka.